server/routes/viewReport.test.ts

```diff
@@ -0,0 +1,227 @@
+  userService.getUsers.mockResolvedValueOnce(users)
+
+  const prisons = {
+    LEI: leeds,
+    MDI: moorland,
+  }
+  prisonApi.getPrisons.mockResolvedValue(prisons)
+
+  const prisoners: Record<string, OffenderSearchResult> = { A1111AA: andrew }
+  offenderSearchApi.getPrisoners.mockResolvedValue(prisoners)
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('GET view report page with details', () => {
+  beforeEach(() => {
+    const now = new Date(2023, 11, 5, 12, 34, 56)
+    const mockedReport = convertReportWithDetailsDates(
+      mockReport({ reportReference: '6543', reportDateAndTime: now, withDetails: true }),
+    )
+
+    incidentReportingApi.getReportWithDetailsById.mockResolvedValueOnce(mockedReport)
+  })
+  it('should render report page with all sections', () => {
+    return request(app)
+      .get('/reports/6543')
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Report 6543')
+        expect(res.text).toContain('Incident reported in Moorland (HMP &amp; YOI) by John Smith')
+        expect(res.text).toContain('Incident details')
+        expect(res.text).toContain('Incident type')
+        expect(res.text).toContain('Prisoners involved')
+        expect(res.text).toContain('Staff involved')
+        expect(res.text).toContain('Question responses')
+        expect(res.text).toContain('Correction requests')
+        expect(incidentReportingApi.getReportWithDetailsById).toHaveBeenCalledTimes(1)
+      })
+  })
+  it('should render incident details', () => {
+    return request(app)
+      .get('/reports/6543')
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Date and time of incident')
+        expect(res.text).toContain('5 December 2023, 11:34')
+        expect(res.text).toContain('Description')
+        expect(res.text).toContain('A new incident created in the new service of type FINDS')
+      })
+  })
+  it('should render incident type with correct formatting', () => {
+    return request(app)
+      .get('/reports/6543')
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Incident type')
+        expect(res.text).toContain('Finds')
+      })
+  })
+  it('should render prisoners involved with roles and outcomes. Names correct if available', () => {
+    return request(app)
+      .get('/reports/6543')
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Andrew Arnold')
+        expect(res.text).toContain('Active involvement - Investigation (local)')
+        expect(res.text).toContain('A2222BB')
+        expect(res.text).toContain('Suspected involved - No outcome')
+      })
+  })
+  it('should render staff involved with roles', () => {
+    return request(app)
+      .get('/reports/6543')
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('lev79n')
```

Comment:
ah this should turn into a staff member's name! you should add a few more entries to the `const users: Record<string, User> = …` by referencing the fake users in /server/data/testData/prisonApi.ts

Reply:
Have included this now. I just took the name and username from the test data staff member. Did go to import it but it is a different structure to what is returned by the `getUsers` function. So I just manually added it to the mock result instead.